### PR TITLE
Initial work on upgrading OpenWPM to Firefox ESR 60

### PR DESCRIPTION
**Pull request by motin**
Work in progress, not ready for merge yet. Created for discussion and ongoing review purposes. 

Relates to https://github.com/citp/OpenWPM/issues/148

@englehardt In order to make this a collaborative effort, please create a `firefox-upgrade` or `develop` branch in the OpenWPM repo for this PR to target, so that the master branch can still be used for bugfixes to the stable version while we get the major version upgrade in order. 

(Note: this PR includes the commits from https://github.com/citp/OpenWPM/pull/219 as well. It is advised to first review and merge that PR, whereafter this PR will be rebased on the current master branch, making it easier to review)
EDIT: https://github.com/citp/OpenWPM/pull/219 has been merged and this PR has been rebased.

### CODE REVIEW

**Comment by motin** [comment]
(Note that travis is not be able to install the proper selenium package due to https://github.com/SeleniumHQ/selenium/issues/6470 - thus all travis tests fail until the upstream issue is fixed or a workaround is implemented)

File: automation/Extension/firefox/webextension/manifest.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "content_scripts": [
+    {
+      "matches": [
+        "<all_urls>"
+      ],
+      "js": [
+        "content.js"
+      ],
+      "run_at": "document_start",
+      "all_frames": true
```
[the remark this answers is not in the document]
Comment (motin) [reply automation/Extension/firefox/webextension/manifest.json]: Very good point

File: test/test_pages/js_instrument/framed1.html [hunk 2]
```diff
@@ -94,10 +94,29 @@ <h3>Test page frame for JS Instrument's instrumentObject method</h3>
...
         console.log("nestedObj call 1");
       }
-      // Instrument object non-recursively
-      console.log("Instrumenting window.frame1Test!");
-      window.instrumentObject(window.frame1Test, "window.frame1Test");
-      interactWithTestObjects();
+
+      // Wait for window.instrumentObject to be set by
+      // the injected page script before continuing
+      const instrumentObjectAvailable = new Promise(function(resolve) {
```
[the remark this answers is not in the document]
Comment (motin) [reply test/test_pages/js_instrument/framed1.html]: Currently, there is no way for webextensions to inject content scripts before any content is loaded, so there is always a race condition between the page script and the injected page script, hence this "wait" for the instrumentation to be available before attempting to use it. Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1378459

**Comment by motin** [comment]
Rebased and ready for merge. Also including the latest commits to master, so that the develop branch is up to date.

File: test/test_pages/js_instrument/framed1.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (englehardt) [reply test/test_pages/js_instrument/framed1.html]: Ah, if I'm understanding Bug 1378459 correctly, the issue is related to extension startup and thus will be present on the *first* page load only right? I suspect we can have the browser manager process block on a signal from the extension before submitting the first top-level request. I'll file a new bug for this.

**Pull request merged**